commands/slash/openstack.go

```diff
@@ -0,0 +1,280 @@
+package slash
+
+import (
+	"bytes"
+	"fmt"
+	"os"
+	"os/exec"
+	"strings"
+
+	"github.com/bwmarrin/discordgo"
+	"github.com/ritsec/ops-bot-iii/commands/slash/permission"
+	"github.com/ritsec/ops-bot-iii/config"
+	"github.com/ritsec/ops-bot-iii/data"
+	"github.com/ritsec/ops-bot-iii/helpers"
+	"github.com/ritsec/ops-bot-iii/logging"
+	"gopkg.in/DataDog/dd-trace-go.v1/ddtrace/tracer"
+)
+
+var (
+	// Environment variables required for openstack CLI
+	OS_AUTH_URL             string = config.GetString("openstack.ENV.OS_AUTH_URL")
+	OS_PROJECT_ID           string = config.GetString("openstack.ENV.OS_PROJECT_ID")
+	OS_PROJECT_NAME         string = config.GetString("openstack.ENV.OS_PROJECT_NAME")
+	OS_USER_DOMAIN_NAME     string = config.GetString("openstack.ENV.OS_USER_DOMAIN_NAME")
+	OS_PROJECT_DOMAIN_ID    string = config.GetString("openstack.ENV.OS_PROJECT_DOMAIN_ID")
+	OS_USERNAME             string = config.GetString("openstack.ENV.OS_USERNAME")
+	OS_PASSWORD             string = config.GetString("openstack.ENV.OS_PASSWORD")
+	OS_REGION_NAME          string = config.GetString("openstack.ENV.OS_REGION_NAME")
+	OS_INTERFACE            string = config.GetString("openstack.ENV.OS_INTERFACE")
+	OS_IDENTITY_API_VERSION string = config.GetString("openstack.ENV.OS_IDENTITY_API_VERSION")
+
+	// Paths for scripts to automate openstack user management
+	new_member      string = config.GetString("openstack.SCRIPTS.new_member")
+	reset_password  string = config.GetString("openstack.SCRIPTS.reset_password")
+	check_if_exists string = config.GetString("openstack.SCRIPTS.check_if_exists")
+)
+
+func Openstack() (*discordgo.ApplicationCommand, func(s *discordgo.Session, i *discordgo.InteractionCreate)) {
+	return &discordgo.ApplicationCommand{
+			Name:                     "openstack",
+			Description:              "Create or reset your openstack account",
+			DefaultMemberPermissions: &permission.Member,
+			Options: []*discordgo.ApplicationCommandOption{
+				{
+					Type:        discordgo.ApplicationCommandOptionString,
+					Name:        "option",
+					Description: "Option of create or reset",
+					Required:    true,
+					Choices: []*discordgo.ApplicationCommandOptionChoice{
+						{
+							Name:  "Create",
+							Value: "Create",
+						},
+						{
+							Name:  "Reset",
+							Value: "Reset",
+						},
+					},
+				},
+			},
+		},
+		func(s *discordgo.Session, i *discordgo.InteractionCreate) {
+			span := tracer.StartSpan(
+				"commands.slash.openstack:Openstack",
+				tracer.ResourceName("/openstack"),
+			)
+			defer span.Finish()
+
+			ssOption := i.ApplicationCommandData().Options[0].StringValue()
+			err := helpers.InitialMessage(s, i, fmt.Sprintf("You ran the /openstack command to %s your account!", strings.ToLower(ssOption)))
+			if err != nil {
+				logging.Error(s, err.Error(), i.Member.User, span)
+			}
+
+			// Initialize the environment variables for Openstack CLI
+			SetOpenstackRC()
```

Comment:
Since these variables are initially stored in the OBIII config file, I'd probably move this to the `main.go` file as it can be seen as redundant to keep setting these on every Openstack command. However, I do see the argument where the host's environment variables could get overwritten after OBIII's initial startup

Reply:
Depending on gophercloud's implementation, I may or may not need it but in case if I do, I will move it to main.go.